pandas-stubs/core/indexes/interval.pyi

```diff
@@ -259,31 +262,31 @@ class IntervalIndex(IntervalMixin, ExtensionIndex, Generic[IntervalT]):
     @overload
     def __gt__(self, other: pd.Series[IntervalT]) -> pd.Series[bool]: ...  # type: ignore[misc]
     @overload
-    def __gt__(self, other: object) -> Literal[False]: ...
+    def __gt__(self, other: object) -> Never: ...
     @overload  # type: ignore[override]
     def __ge__(self, other: IntervalT | IntervalIndex[IntervalT]) -> np_ndarray_bool: ...  # type: ignore[misc]
     @overload
     def __ge__(self, other: pd.Series[IntervalT]) -> pd.Series[bool]: ...  # type: ignore[misc]
     @overload
-    def __ge__(self, other: object) -> Literal[False]: ...
+    def __ge__(self, other: object) -> Never: ...
     @overload  # type: ignore[override]
     def __le__(self, other: IntervalT | IntervalIndex[IntervalT]) -> np_ndarray_bool: ...  # type: ignore[misc]
     @overload
     def __le__(self, other: pd.Series[IntervalT]) -> pd.Series[bool]: ...  # type: ignore[misc]
     @overload
-    def __le__(self, other: object) -> Literal[False]: ...
+    def __le__(self, other: object) -> Never: ...
     @overload  # type: ignore[override]
     def __lt__(self, other: IntervalT | IntervalIndex[IntervalT]) -> np_ndarray_bool: ...  # type: ignore[misc]
     @overload
     def __lt__(self, other: pd.Series[IntervalT]) -> bool: ...  # type: ignore[misc]
     @overload
-    def __lt__(self, other: object) -> Literal[False]: ...
-    @overload  # type: ignore[override]
+    def __lt__(self, other: object) -> Never: ...
+    @overload
     def __eq__(self, other: IntervalT | IntervalIndex[IntervalT]) -> np_ndarray_bool: ...  # type: ignore[misc]
     @overload
     def __eq__(self, other: pd.Series[IntervalT]) -> pd.Series[bool]: ...  # type: ignore[misc]
     @overload
-    def __eq__(self, other: object) -> Literal[False]: ...
+    def __eq__(self, other: object) -> Never: ...
```

Comment:
This should remain `Literal[False]`, because it is always valid code to do equality against another object.